docs/_tooling/extensions/score_metamodel/tests/test_rules_file_based.py

```
@@ -0,0 +1,137 @@
+        for no, line in enumerate(f, start=1):
+            if line.startswith(".. "):  # Beginning of new need
+                if test_info:
+                    test_info.lineno = no
+                    statements.append(test_info)
+                test_info = None
+            elif line.startswith("#EXPECT:") or line.startswith("#EXPECT-NOT:"):
+                if test_info is None:
+                    test_info = InfoElement()
+                target_list = (
+                    test_info.expected
+                    if line.startswith("#EXPECT:")
+                    else test_info.not_expected
+                )
+                target_list.append(extract_warning(line))
+        # Check last InfoElement
+        if test_info:
+            print("ERROR: Teststatement without according need found")
+        return statements
+    return None
+
+
+@pytest.mark.parametrize("rst_file", RST_FILES)
+def test_check_rules(
+    rst_file: str, sphinx_app_setup: Callable[[Path], SphinxTestApp]
+) -> None:
+    assert (
+        test_data := extract_test_data(RST_DIR / rst_file)
+    ), "Unable to extract test data"
+    app: SphinxTestApp = sphinx_app_setup(RST_DIR / rst_file)
+    try:
+        os.chdir(app.srcdir)  # Change working directory to the source directory
+        app.build()
+        warn_text: str = app.warning.getvalue()
+        for test in test_data:
+            for expected in test.expected:
+                assert (
+                    f"{Path(rst_file).name}:{test.lineno}: WARNING: {expected}"
+                    in warn_text
+                ), f"Expected warning: {[expected]} not found"
+            for not_expected in test.not_expected:
+                assert (
+                    f"{Path(rst_file).name}:{test.lineno}: WARNING: {not_expected}"
+                    not in warn_text
+                ), f"Unexpected warning: {[not_expected]} found"
+    except Exception as e:
+        assert False, f"Build failed: {e}"
```

I'm still unsure if an assert is better here or if it's better to raise the exception in a controlled way like: 
`raise e`

---

As that might behave nicer with pyright